doc/release/multilocale.rst

```diff
@@ -4,233 +4,198 @@
+    (see :ref:`Using the Portable UDP Conduit <using-udp>`)
 
-   .. code-block:: bash
+See the GASNet documentation for more information on each of these conduits.
 
-     chpl -o hello6-taskpar-dist $CHPL_HOME/examples/hello6-taskpar-dist.chpl
+Current defaults are:
 
+====================  ===================
+CHPL_TARGET_PLATFORM  CHPL_COMM_SUBSTRATE
+====================  ===================
+cray-cs                ibv
+cray-xc                aries
+cray-xe                gemini
+cray-xk                gemini
+cray-xt                mpi
+pwr5                   lapi
+pwr6                   ibv
+other                  udp
+====================  ===================
 
-#. When you compile a Chapel program for multiple locales, you should
-   typically see two binaries (e.g., hello6-taskpar-dist and
-   hello6-taskpar-dist_real).  The first binary contains code to
-   launch the Chapel program onto the compute nodes using a GASNet run
-   command.  The second contains the program code itself.  You can use
-   the ``-v`` flag to see the commands used to launch your program.  See
-   :ref:`readme-launcher` for further details.
+.. _set-comm-segment:
 
+Setting CHPL_GASNET_SEGMENT
++++++++++++++++++++++++++++
 
-#. Multi-locale executions require the number of locales to be
-   specified on the command line.  Other than this, execute your
-   Chapel program as usual.  For example:
+Advanced GASNet users can set ``CHPL_GASNET_SEGMENT`` to choose a memory
+segment to use with GASNet. A GASNet segment is a region of memory that
+is expected to be used for remote memory access. The GASNet library works
+to make memory in this segment is available for accelerated memory access
+supported directly by network hardware.  The options are:
 
-   .. code-block:: bash
+everything
+  All memory is available for remote memory access
+fast
+  A limited portion of memory is available and optimized for fastest remote
+  memory access
+large
+  As with fast, but a larger amount of memory is available for communication
 
-     ./hello6-taskpar-dist -nl 2
+Current defaults are:
+
+===================  ====================
+CHPL_COMM_SUBSTRATE  CHPL_GASNET_SEGMENT
+===================  ====================
+aries                fast
+gemini               fast
+ibv                  large
+portals              fast
+other                everything
+===================  ====================
 
+Troubleshooting
++++++++++++++++
+
+If you are trying to debug job launch, try adding ``-v`` to your
+program's command line and set:
+
+  .. code-block:: bash
+
+    export GASNET_VERBOSEENV=1
+
+We've had best results with console I/O using:
+
+  .. code-block:: bash
+
+    export GASNET_ROUTE_OUTPUT=0
+
+but your mileage may vary.
+
+.. _set-comm-debugging:
+
+Advanced users may want to set ``CHPL_COMM_DEBUG`` in order to enable
+GASNet's internal sanity checking. (It is off by default.)
+Note that you'll need to re-make the compiler and runtime when changing
```

Comment:
I think it makes more sense to say "re-make Chapel" since I'm not so sure drawing a fine distinction between the compiler and runtime is actually useful in this document.